### ISSUE TITLE
tag field in FractalClient.query_results

### ISSUE THREAD

**Issue by mattwelborn** [MolSSI/QCFractal#392]
**Is your feature request related to a problem? Please describe.**
I would like to be able to list all Results associated with a given tag on a multi-user/multi-project instance like the QCArchive

**Describe the solution you'd like**
A tag field in FractalClient.query_results

**Describe alternatives you've considered**
Pulling all results and then filtering the tag client-side. This takes a while and likely puts unnecessary load on the server. 

**Additional context**
This may also be useful when we build a dashboard.

**Comment by mattwelborn** [comment]
What I'm really trying to do is manage large numbers of calculations while building a collection. Broadly speaking, what I would like to do is:

1. See how many jobs I have waiting in the queue
2. See how many (and which jobs) have failed
3. Examine and restart failed jobs
4. Examine a collection to see how complete it is

My current solution to problem 1 is to keep track of task ids when I submit them with `FractalClient.add_compute` (`Dataset.compute` does not return all ids if there are more than `FractalClient.query_limit` molecules in the dataset) and then check them with `FractalClient.query_results`. 

For items 2 and 3, I iterate through the results table looking for jobs that match my manager(s):

```
chunk_size = client.query_limit

all_fails = []
n_restarted = 0
skip = 0
while True:
    print(skip)
    fails = client.query_results(status='ERROR', skip=skip)
    if len(fails) == 0:
        break
    all_fails += [fail for fail in fails if fail.manager_name=="MolSSI_ARC_Cascades_Parsl-calogin1-435ab7a2-33f9-45f8-8012-09b2d1a4a9e5"]
    skip += chunk_size
    
n_error = len(all_fails)
print("Total errors", n_error)

for i in range(0,len(all_fails),chunk_size):
    ie = min(i+chunk_size,len(all_fails))
    n_restarted += client.modify_tasks('restart', [fail.id for fail in all_fails[i:ie]]).n_updated
    print(n_restarted)
```
This seems inefficient, and doesn't work if the jobs get stuck without being assigned a manager as in issue #391 .

For the fourth item, I iterate through the dataset history looking for NaNs. For example, to count failed/incomplete jobs:

```
for method, basis in ds.list_history().reset_index().groupby(['method','basis']).size().index:
    index = qm7.query(method=method, basis=basis)
    print(index, ds.df.reset_index()[index].isna().sum())
```

**Comment by dgasmith** [comment]
We could provide data like this:
```
tag     priority status count
mopac	1	waiting	36
mopac	1	error	62
mgwarc	1	running	530
mgwarc	1	waiting	183468
```

Then perhaps a `client.query_tasks(tag='mopac', status='error', manager_name='MolSSI_ARC_Cascades_Parsl', ...)`  the task `base_result_id` would link back to the record's error flag.

In the future we can do something like `client.get_errors(tag=...)`?

**Comment by mattwelborn** [comment]
That makes sense to me. As I understand it, the query_tasks function would need to add `tag` and `manager` arguments?

**Comment by dgasmith** [comment]
Yes, this should be rather straight forward to add to `sqlalchemy_sockets.py`, `rest_models.py:TaskQueueGETBody`, and `client.py`. While this unfortunately occurs in many places it is a matter of propagating the new field through.

You can view `sql_models.py:TaskQueueORM` for all queryable fields.

**Comment by dgasmith** [comment]
Closed by #394.
